packages/amplication-server/src/core/resource/resource.service.ts

```diff
@@ -51,9 +57,40 @@ export class ResourceService {
     private blockService: BlockService,
     private environmentService: EnvironmentService,
     private buildService: BuildService,
-    private serviceSettingsService: ServiceSettingsService
+    private serviceSettingsService: ServiceSettingsService,
+    private readonly projectConfigurationSettingsService: ProjectConfigurationSettingsService
   ) {}
 
+  private async assertFirstProjectConfiguration(
+    projectId: string
+  ): Promise<void> {
+    const projectConfiguration = await this.prisma.resource.findFirst({
+      where: { projectId, resourceType: EnumResourceType.ProjectConfiguration }
+    });
+    if (!isEmpty(projectConfiguration)) {
+      throw new ProjectConfigurationExistError();
+    }
+    return;
+  }
+
+  async createProjectConfiguration(
+    projectId: string,
+    userId: string
+  ): Promise<Resource> {
+    await this.assertFirstProjectConfiguration(projectId);
+    const resource = await this.prisma.resource.create({
+      data: {
+        color: DEFAULT_RESOURCE_COLORS.projectConfiguration,
+        resourceType: EnumResourceType.ProjectConfiguration,
+        description: '',
+        name: 'Project configuration',
```

Comment:
Needs to be capitalized like a title - `Project Configuration` instead of `Project configuration`

Reply:
Indeed, let's put this in a constant.